src/ClockPanel.tsx

```diff
@@ -258,71 +215,60 @@ class UnthemedClockPanel extends PureComponent<Props, State> {
     );
   }
 
-  renderDate() {
-    const { now } = this.state;
-    const { dateSettings } = this.props.options;
+  function renderDate() {
+    const { dateSettings } = props.options;
 
     const className = css`
       font-size: ${dateSettings.fontSize};
       font-weight: ${dateSettings.fontWeight};
     `;
 
     const display = now.locale(dateSettings.locale || '').format(dateSettings.dateFormat);
-    
+
     return (
       <span>
         <h3 className={className}>{display}</h3>
       </span>
     );
   }
 
-  renderTime() {
-    const { now } = this.state;
-    const { options } = this.props;
-    const { timeSettings, mode } = options;
+  function renderTime() {
+    const { timeSettings, clockType, mode } = props.options;
 
     const className = css`
       font-size: ${timeSettings.fontSize};
       font-weight: ${timeSettings.fontWeight};
     `;
 
-    let display = now.format(this.getTimeFormat());
-    if (mode === ClockMode.countdown) {
-      display = this.getCountdownText();
-    } else if (mode === ClockMode.countup) {
-      display = this.getCountupText();
+    let display = '';
+    switch (mode) {
+      case ClockMode.countdown:
+        display = getCountdownText();
+        break;
+      case ClockMode.countup:
+        display = getCountupText();
+        break;
+      default:
+        display = now.format(getTimeFormat(clockType, timeSettings));
+        break;
     }
 
     return <h2 className={className}>{display}</h2>;
   }
 
-  render() {
-    const { options, width, height, theme } = this.props;
-    const { dateSettings, timezoneSettings, bgColor } = options;
-
-    const className = css`
-      display: flex;
-      align-items: center;
-      justify-content: center;      
-      flex-flow: column wrap;
-      text-align: center;
-      background-color: ${!bgColor ? theme.colors.background.primary : theme.v1.visualization.getColorByName(bgColor)};
-    `;
-
-    return (
-      <div
-        className={className}
-        style={{
-          width,
-          height,
-        }}
-      >
-        {dateSettings.showDate && this.renderDate()}
-        {this.renderTime()}
-        {timezoneSettings.showTimezone && this.renderZone()}
-      </div>
-    );
-  }
+  return (
+    <div
+      className={className}
+      style={{
+        width,
+        height,
+      }}
+    >
+      {dateSettings.showDate && renderDate()}
+      {renderTime()}
+      {timezoneSettings.showTimezone && renderZone()}
+    </div>
+  );
 }
 
-export const ClockPanel = withTheme2(UnthemedClockPanel);
+export const ClockPanel = withTheme2(UnthemedClockPanelFunctional);
```

Comment:
nit: Instead of wrapping the component in a HOC you can use the `useTheme2` hook directly in the component to get the theme.